components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -8,15 +8,309 @@
 #include <string>
 #include <utility>
 
+#include "brave/common/brave_wallet_constants.h"
+#include "base/base64.h"
+#include "base/bind.h"
+#include "base/files/file_util.h"
 #include "base/task/post_task.h"
 #include "base/task_runner_util.h"
-#include "brave/components/brave_wallet/browser/brave_wallet_controller.h"
+#include "brave/common/extensions/extension_constants.h"
+#include "brave/components/brave_wallet/browser/browser_wallet_delegate.h"
+#include "components/user_prefs/user_prefs.h"
 #include "content/public/browser/browser_context.h"
+#include "crypto/aead.h"
+#include "crypto/hkdf.h"
+#include "crypto/random.h"
+#include "crypto/symmetric_key.h"
+#include "brave/common/pref_names.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/common/url_constants.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/browser/shared_user_script_master.h"
 
-BraveWalletService::BraveWalletService(content::BrowserContext* context)
+namespace {
+  bool ResetCryptoWalletsOnFileTaskRunner(
+      const base::FilePath& path) {
+    return base::DeleteFile(path, true);
+  }
+}
+
+BraveWalletService::BraveWalletService(content::BrowserContext* context,
+    std::unique_ptr<BrowserWalletDelegate> browser_wallet_delegate)
     : context_(context),
-      controller_(new BraveWalletController(context)),
-      weak_factory_(this) {
+      browser_wallet_delegate_(std::move(browser_wallet_delegate)),
+      extension_registry_observer_(this),
+      file_task_runner_(base::CreateSequencedTaskRunner(
+          {base::ThreadPool(), base::MayBlock(),
+           base::TaskPriority::BEST_EFFORT,
+           base::TaskShutdownBehavior::BLOCK_SHUTDOWN})),
+    weak_factory_(this) {
+  pref_change_registrar_ = std::make_unique<PrefChangeRegistrar>();
+  pref_change_registrar_->Init(user_prefs::UserPrefs::Get(context_));
```

Comment:
fyi - this is ok, but the typical pattern is to pass in the `PrefService` from `profile->GetPrefs`

Reply:
next time ;)